docs/proposal-007-instance-selector.md

```diff
@@ -0,0 +1,269 @@
+
+```yaml
+instanceSelector:
+  vCPUs: <integer>
+  memory: <integer>
+```
+
+At launch only the `vCPUs` and `memory` options will be supported, additional options may be added later based on community feedback.
+
+**Note:** Because fields are case-sensitive, `vCPUs` may be difficult to get right, especially for users unfamiliar with Go naming conventions. However, `vCPUs` with the same case is standard enough that it's widely used by all major cloud providers. `vcpus` is easier to remember but it's inconsistent with the other field names.
+
+### eksctl create cluster
+
+When `eksctl create cluster` is called with the instance selector options and `--dry-run`, eksctl will output a ClusterConfig file containing a nodegroup representing the CLI options and the instance types set to the instances matched by the instance selector resource criteria.
+
+```shell
+$ eksctl create cluster --name development --managed --instance-selector-vcpus=2 --instance-selector-memory=4 --dry-run
+
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: development
+  region: us-west-2
+
+managedNodeGroups:
+- name: <name>
+  instanceTypes: ["c5.large", "c5a.large", "c5ad.large", "c5d.large", "t2.medium", "t3.medium", "t3a.medium"]
+  instanceSelector:
+    vCPUs: 2
+    memory: 4
+
+```
+
+The `instanceSelector` field representing the CLI options will also be added to the ClusterConfig file for visibility and documentation purposes. When `--dry-run` is omitted, this field will be ignored and the `instanceTypes` field will be used, otherwise any changes to `instanceTypes` would get overridden by eksctl.
+
+For self-managed nodegroups, `instancesDistribution.instanceTypes` will be set:
+
+```shell
+$ eksctl create cluster --name development --instance-selector-vcpus=2 --instance-selector-memory=4 --dry-run
+
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: development
+  region: us-west-2
+
+nodeGroups:
+- name: <name>
+  instancesDistribution:
+    instanceTypes: ["c5.large", "c5a.large", "c5ad.large", "c5d.large", "t2.medium", "t3.medium", "t3a.medium"]
+  instanceSelector:
+    vCPUs: 2
+    memory: 4
+
+
+```
+
+
+If `--dry-run` is omitted, eksctl will proceed to creating the cluster and nodegroup with the instance types matching the instance selector resource criteria. 
+
+---
+
+When a ClusterConfig file is passed with `--dry-run`, eksctl will output a ClusterConfig file containing the same set of nodegroups after expanding each nodegroup's instance selector resource criteria.
+
+
+```shell
+$ cat cluster.yaml
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: cluster
+  region: us-west-2
+
+managedNodeGroups:
+- name: linux
+  instanceSelector:
```

Comment:
is it confusing that this is only in the MNG section? How do we handle merging MNG and non-MNG groups?

Reply:
I'm not sure I follow. This field is also present in the `nodeGroups` section below: https://github.com/weaveworks/eksctl/pull/3107/files#diff-af7ec3d70c271157e75ad67f9bb60c75c8596ace0bee45c3b6a83d7debd9408cR126

>How do we handle merging MNG and non-MNG groups?

Can you expand on this?

Reply:
you answered my question -must had missed it on the first pass.